src/js/node/stream.js

```diff
@@ -2608,7 +2608,8 @@ var require_readable = __commonJS({
     function addChunk(stream, state, chunk, addToFront) {
       $debug("adding chunk", stream.__id);
       $debug("chunk", chunk.toString(), stream.__id);
-      if (state.flowing && state.length === 0 && !state.sync && stream.listenerCount("data") > 0) {
+      const has_listenners = stream.listenerCount("data") > 0;
```

Comment:
This code should very much mirror what Node.js does.

Is this what Node.js does?

Reply:
let me check that

Reply:
Reverted it back in https://github.com/oven-sh/bun/pull/6487/commits/b68c074aa49d5f6ec9c9659bf9791d01f5c6bea3